framework/DataObjects/XDataSet.py

```diff
@@ -689,11 +689,17 @@ def _checkAlignedIndexes(self,rlz,tol=1e-15):
     for index in self.indexes:
       # if it's aligned so far, check if it still is
       if index in self._alignedIndexes.keys():
-        # if close enough, then keep the aligned values; otherwise, take action
-        if isinstance(rlz[index][0],(float,int)):
-          closeEnough = all(np.isclose(rlz[index],self._alignedIndexes[index],rtol=tol))
+        # first, if lengths don't match, they're not aligned.
+        # TODO there are concerns this check may slow down runs; it should be profiled along with other bottlenecks to optimize our efforts.
+        if len(rlz[index]) != len(self._alignedIndexes[index]):
+          closeEnough = False
         else:
-          closeEnough = all(rlz[index] == self._alignedIndexes[index])
+          # "close enough" if float/int, otherwise require exactness
+          if isinstance(rlz[index][0],(float,int)):
```

Comment:
what about it is a numpy float or int? 
You should add ```numpy.integer``` and ```numpy.number``` for checking all of them

Reply:
I do have an issue when the ```dtype=np.int32``` in the past.

Reply:
```python
a = np.array([6],dtype=np.integer)
type(a[0])           # result is <type 'numpy.int64'>
isinstance(a[0],int) # result is True

a = np.array([3.14],dtype=np.float)
type(a[0])             # result is <type 'numpy.float64'>
isinstance(a[0],float) # result is True
```
Inheritance! 😃

Reply:
Can you test ```np.int32, np.float32, np.int8, ...```?

Reply:
Sorry, why are we using a variety of integer types?  Is this because of different numpy installations?

Reply:
I have no idea about this, but I do have a case (data mining) when the return type is ```np.int32```, and the code crashes. If I remembered correctly, the type of returned labels is ```np.int32```.

Reply:
It is not ```Inheritance```
```np.int64``` and ```np.float64``` are implicitly mapped as python Floats and Integer. I am not surprised your ```isinstance``` works fine with the default ```np.integer``` and ```np.float```.

Indeed : 
```
>>> import numpy as np
>>> a = np.array([6],dtype=np.float32)
>>> type(a[0])
<type 'numpy.float32'>
>>> isinstance(a[0],float)
```

Reply:
Again, the point is, if we're going to have some dozen types that have no common root but all need to be considered floats, we need a more intelligent way throughout the data objects (and raven in general) to make this kind of a check.  It better be a fast method, too, because looping over the options will be painful.

Is checking against float,int,np.integer,np.float sufficient?  No, because as I just tested, while all the np.int* are np.integer types, not all np.float* are np.float types.  This bears more discussion than comments on this PR merit.

Reply:
Paul.
Firstly I told you that the correct way to check for any float-like types in numpy is ```numpy.number``` (from which all the numpy.float* inherits from) and not ```numpy.float```
Second, I agree that this checks should be handled in the utilities where we are going to have control on the additional ```types``` we would (if any) need in the future.


```
>>> a = np.array([6],dtype=np.float32)
>>> b = np.array([6],dtype=np.float64)
>>> c = np.array([6],dtype=np.float128)

>>> isinstance(a[0],np.number)
True
>>> isinstance(b[0],np.number)
True
>>> isinstance(c[0],np.number)
True
```

Reply:
From NumPy: 
<img width="522" alt="screen shot 2018-01-16 at 12 11 05 pm" src="https://user-images.githubusercontent.com/7321071/35007273-63fa7532-fab6-11e7-8231-f6c25bc27d5d.png">

Reply:
These are the fundamental data types from NumPy

Reply:
Please, do not add more here.  This needs to be a seperate discussion from this PR.

Reply:
This is handled in #524.